### PR TITLE
create and update posture policy | SSPROD-39113 

### PR DESCRIPTION
**Pull request by hila1608**

### CODE REVIEW

**Review by yaminSapir — APPROVED** [review]

File: sysdig/internal/client/v2/model.go [hunk 1]
```diff
@@ -926,21 +926,89 @@ type PosturePolicyZoneMeta struct {
...
+	ID                string                  `json:"id,omitempty"`
+	Name              string                  `json:"name,omitempty"`
+	Type              int                     `json:"type,omitempty"`
+	Kind              int                     `json:"kind,omitempty"`
+	Description       string                  `json:"description,omitempty"`
+	Version           string                  `json:"version,omitempty"`
+	Link              string                  `json:"link,omitempty"`
+	Authors           string                  `json:"authors,omitempty"`
+	PublishedData     string                  `json:"publishedDate,omitempty"`
+	RequirementsGroup []RequirementsGroup     `json:"requirementFolders,omitempty"`
```
Comment (filiptubic) [inline sysdig/internal/client/v2/model.go:938]: Should this json field be called `requirementFolder` or `requirementGroup`

Comment (filiptubic) [reply sysdig/internal/client/v2/model.go:938]: it has different name in json annotation, might be mistake, just checking

File: sysdig/internal/client/v2/model.go [hunk 2]
```diff
@@ -926,21 +926,89 @@ type PosturePolicyZoneMeta struct {
...
+	MinKubeVersion    float64                 `json:"minKubeVersion,omitempty"`
+	MaxKubeVersion    float64                 `json:"maxKubeVersion,omitempty"`
+	IsCustom          bool                    `json:"isCustom,omitempty"`
+	IsActive          bool                    `json:"isActive,omitempty"`
+	Platform          string                  `json:"platform,omitempty"`
+	Zones             []PosturePolicyZoneMeta `json:"zones,omitempty"`
+}
+
+type RequirementsGroup struct {
+	Id                        string              `json:"id,omitempty"`
```
Comment (filiptubic) [inline sysdig/internal/client/v2/model.go:948]: Capitalize both letters so `Id` to `ID`

File: sysdig/internal/client/v2/model.go [hunk 3]
```diff
@@ -926,21 +926,89 @@ type PosturePolicyZoneMeta struct {
...
+	Name                      string                    `json:"name,omitempty"`
+	Requirements              []CreateRequirement       `json:"requirements,omitempty"`
+	Description               string                    `json:"description,omitempty"`
+	Authors                   string                    `json:"author,omitempty"`
+	Folders                   []CreateRequirementsGroup `json:"groups,omitempty"`
+	RequirementFolderParentID string                    `json:"requirementFolderParentId,omitempty"`
+}
+
+type CreateRequirement struct {
+	Id                  string                     `json:"id,omitempty"`
```
Comment (filiptubic) [inline sysdig/internal/client/v2/model.go:997]: Id -> ID

File: sysdig/internal/client/v2/posture_policies.go [hunk 4]
```diff
@@ -7,14 +7,18 @@ import (
 )
 
 const PosturePolicyListPath = "%s/api/cspm/v1/policy/policies/list"
+const PosturePolicyCreatePath = "%s/api/cspm/v1/policy"
+const PosturePolicyGetPath = "%s/api/cspm/v1/policy/policies/view/%d"
```
Comment (filiptubic) [inline sysdig/internal/client/v2/posture_policies.go:12]:
```suggestion
const (
    PosturePolicyListPath = "%s/api/cspm/v1/policy/policies/list"
    PosturePolicyCreatePath = "%s/api/cspm/v1/policy"
    PosturePolicyGetPath = "%s/api/cspm/v1/policy/policies/view/%d"
)
```

File: sysdig/internal/client/v2/posture_policies.go [hunk 5]
```diff
@@ -27,7 +31,46 @@ func (client *Client) ListPosturePolicies(ctx context.Context) ([]PosturePolicy,
...
+	payload, err := Marshal(p)
+	if err != nil {
+		return nil, "", err
+	}
+	response, err := client.requester.Request(ctx, http.MethodPost, client.getPosturePolicyURL(PosturePolicyCreatePath), payload)
+	if err != nil {
+		return nil, "", err
+	}
+	defer response.Body.Close()
+	if response.StatusCode != http.StatusOK && response.StatusCode != http.StatusCreated && response.StatusCode != http.StatusAccepted {
```
Comment (filiptubic) [inline sysdig/internal/client/v2/posture_policies.go:44]: why checking all those status codes? Do you know what is the status code of the service?

File: sysdig/resource_sysdig_secure_posture_policy.go [hunk 6]
```diff
@@ -0,0 +1,430 @@
...
+			groupData["requirement"] = requirementsData
+		}
+		if len(group.Folders) > 0 {
+			nestedGroupsData := setGroups(d, group.Folders)
+			groupData["group"] = nestedGroupsData
+		}
+
+		groupsData = append(groupsData, groupData)
+	}
+	fmt.Println("groupsData: ", groupsData)
```
Comment (filiptubic) [inline sysdig/resource_sysdig_secure_posture_policy.go:319]: remove print from code

File: sysdig/resource_sysdig_secure_posture_policy.go [hunk 7]
```diff
@@ -0,0 +1,430 @@
...
+		ctrlData := map[string]interface{}{
+			"name":    ctrl.Name,
+			"enabled": ctrl.Enabled,
+		}
+		controlsData = append(controlsData, ctrlData)
+	}
+	return controlsData
+}
+func resourceSysdigSecurePosturePolicyDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	// TODO: implement deletion
```
Comment (filiptubic) [inline sysdig/resource_sysdig_secure_posture_policy.go:355]: Is it fine to have non implemented delete?

Comment (hila1608) [reply sysdig/resource_sysdig_secure_posture_policy.go:355]: At the moment, we support only create and update operations. We are currently working on the implementation of delete functionality

File: sysdig/internal/client/v2/posture_policies.go (hunk 5 above)
Comment (hila1608) [reply sysdig/internal/client/v2/posture_policies.go:45]: 201 for create
200 for update
accept status code removed

**Review by filiptubic — APPROVED** [review]

**Review by yaminSapir — APPROVED** [review]

**Review by miguelgordo — APPROVED** [review]

**Pull request merged**